tests/network/test_network_upgrade.py

```diff
@@ -15,3 +24,41 @@ def test_bob_can_follow_treasure_map(enacted_policy_group, ursulas):
 
     bob.follow_treasure_map(enacted_policy_group.treasure_map)
     assert len(bob._ursulas) == len(ursulas)
+
+
+def test_bob_can_issue_a_work_order_to_a_specific_ursula(enacted_policy_group, alice, bob, ursulas):
+    """
+    Now that Bob has his list of Ursulas, he can issue a WorkOrder to one.  Upon receiving the WorkOrder, Ursula
+    saves it and responds by re-encrypting and giving Bob a cFrag.
+
+    This is a multipart test; it shows proper relations between the Characters Ursula and Bob and also proper
+    interchange between a KFrag, PFrag, and CFrag object in the context of REST-driven proxy re-encryption.
+    """
+
+    # We pick up our story with Bob already having followed the treasure map above, ie:
+    assert len(bob._ursulas) == len(ursulas)
+
+    the_pfrag = enacted_policy_group.pfrag
+
+    # We'll test against just a single Ursula - here, we made a WorkOrder for just one.
+    work_orders = bob.generate_work_orders(enacted_policy_group, the_pfrag, num_ursulas=1)
+    assert len(work_orders) == 1
+
+    networky_stuff = MockNetworkyStuff(ursulas)
+
+    ursula_dht_key, work_order = list(work_orders.items())[0]
+    cfrags = bob.get_reencrypted_c_frag(networky_stuff, work_order)
+
+    the_cfrag = cfrags[0]  # We only gave one pFrag, so we only got one cFrag.
```

Comment:
You maybe meant that we have only one Ursula, that's why we've got only one cFrag? I think, number of cfrags is equal to the number of Ursulas?
Or it combines also?